poc-cb-net/cmd/test-client/test-client.go

```diff
@@ -101,67 +151,90 @@ func deployKubernetesCluster(nsID string, mcisID string, vmID string) {
+	pressEnterKeyToContinue(isOn)
+
 	resp, err := client.R().
 		SetHeader("Content-Type", "application/json").
 		SetHeader("Accept", "application/json").
 		SetPathParams(map[string]string{
 			"nsId":   nsID,
 			"mcisId": mcisID,
 		}).
-		SetBody(body).
-		Post("http://localhost:1323/tumblebug/ns/{nsId}/cmd/mcis/{mcisId}")
+		SetQueryParams(map[string]string{
+			"option": "status",
+		}).
+		Get("http://localhost:1323/tumblebug/ns/{nsId}/mcis/{mcisId}")
 
 	// Output print
 	fmt.Printf("\nError: %v\n", err)
 	fmt.Printf("Time: %v\n", resp.Time())
 	fmt.Printf("Body: %v\n", resp)
 
-	fmt.Println("##### End ---------- Special stage - Step 1: Setup environment and tools related to Kubernetes")
+	retMasterVMID := gjson.Get(resp.String(), "status.masterVmId")
+	retVMIDs := gjson.Get(resp.String(), "status.vm.#.id")
+	fmt.Printf("retMasterVMID: %#v\n", retMasterVMID)
+	fmt.Printf("retVMIDs: %#v\n", retVMIDs)
+
+	fmt.Println("##### End ---------- Special stage - Step 1: Retrieve the lead/master VM ID and all VM IDs")
 	fmt.Println("Sleep 5 sec ( _ _ )zZ")
 	time.Sleep(5 * time.Second)
 
-	// Special stage - Step 2: Reboot VMs to apply network configuration
-	// curl -X GET "http://localhost:1323/tumblebug/ns/ns01/control/mcis/mcis01?action=refine" -H "accept: application/json"
-	fmt.Println("\n\n##### Start ---------- Special stage - Step 2: Reboot VMs to apply network configuration")
-	resp, err = client.R().
-		SetHeader("Content-Type", "application/json").
-		SetHeader("Accept", "application/json").
-		SetPathParams(map[string]string{
-			"nsId":   nsID,
-			"mcisId": mcisID,
-		}).
-		SetQueryParams(map[string]string{
-			"action": "reboot",
-		}).
-		Get("http://localhost:1323/tumblebug/ns/{nsId}/control/mcis/{mcisId}")
+	// Special stage - Step 2: Setup environment and tools related to Kubernetes
+	fmt.Println("\n\n##### Start ---------- Special stage - Step 2: Setup environment and tools related to Kubernetes")
+	pressEnterKeyToContinue(isOn)
 
-	// Output print
-	fmt.Printf("\nError: %v\n", err)
-	fmt.Printf("Time: %v\n", resp.Time())
-	fmt.Printf("Body: %v\n", resp)
+	// Set command
+	commandToSetupEnvironmentandTools := `wget https://raw.githubusercontent.com/cloud-barista/cb-larva/develop/scripts/Kubernetes/1.setup-environment-and-tools.sh -O ~/1.setup-environment-and-tools.sh; chmod +x ~/1.setup-environment-and-tools.sh; ~/1.setup-environment-and-tools.sh`
+
+	// Set request body
+	body := fmt.Sprintf(placeHolderBody, commandToSetupEnvironmentandTools)
+	fmt.Printf("body: %#v\n", body)
+
+	// Setup in parallel
+	for _, vmID := range retVMIDs.Array() {
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, vmID string) {
+			defer wg.Done()
+			resp, err = client.R().
+				SetHeader("Content-Type", "application/json").
+				SetHeader("Accept", "application/json").
+				SetPathParams(map[string]string{
+					"nsId":   nsID,
+					"mcisId": mcisID,
+				}).
+				SetBody(body).
+				Post("http://localhost:1323/tumblebug/ns/{nsId}/cmd/mcis/{mcisId}")
```

Comment:
Managing TB endpoint in a variable is even better :) 
endpointTB := "http://localhost:1323" ?